article/app/jobs/MessageUsLifecycle.scala

```diff
@@ -0,0 +1,46 @@
+package jobs
+
+import app.LifecycleComponent
+import common.{AkkaAsync, JobScheduler}
+import play.api.inject.ApplicationLifecycle
+import services.MessageUsService
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+class MessageUsLifecycle(
+    appLifeCycle: ApplicationLifecycle,
+    jobs: JobScheduler,
+    akkaAsync: AkkaAsync,
+    messageUsService: MessageUsService,
+)(implicit
+    ec: ExecutionContext,
+) extends LifecycleComponent {
+
+  appLifeCycle.addStopHook { () =>
+    Future {
+      descheduleJobs()
+    }
+  }
+
+  override def start(): Unit = {
+    descheduleJobs()
+    scheduleJobs()
+
+    // refresh message us data when app starts
+    akkaAsync.after1s {
+      messageUsService.refreshMessageUsData()
+    }
+  }
+
+  private def scheduleJobs(): Unit = {
+    // This job runs every 2 minutes
+    jobs.scheduleEvery("MessageUsAgentRefreshJob", 2.minutes) {
```

Comment:
It probably doesn't make a huge difference in the long run, but how often do we expect the values to change? Does the data need to refreshed every 2 minutes?

Reply:
Thanks for pointing this out @OllysCoding 
They actually don't change quite often, but also when they change, it'd be good if there's not a big delay to appear on the blog. Just talked to our Product manager and he agreed that a 10 minute delay at most could be acceptable. So I set this job to 8 minutes, as there could also be a delay by the back end lambda that's preparing the data and store it in S3. If that's OK with you too?

Reply:
Sounds good to me :)